students/Mintae_K/lesson06/mailroom4.py

```diff
@@ -0,0 +1,98 @@
+# Error Handling function for sending_thank function
+def check_error(amount):
+    try:
+        checked_amount = float(amount)
+        return 1
+    except ValueError:
+        print('\nValue entered is not a number. Only number allowed.\n')
+        return 0
+
+
+# Adding donation to existing person or creating new donation data
+def check_name_exist(f_name, donors):
+    if donors.get(f_name):
+        s = 'Existing donor, please type the donation amount from {}> '
+        return 1, s
+    else:
+        s = 'New donor, please type the donation amount from {}> '
+        return 0, s
+
+
+def append_donation(f_name, amount, yes_exist, donors):
+    #global donors
+    if yes_exist == 1:
+        donors[f_name].append(float(amount))
+    else:
+        donors[f_name] = [float(amount)]
+    return donors
+
+
+# Sending a Thank you
+def sending_thank():
+    f_name = input('Please type the Full Name or list> ')
+    while f_name == 'list':
+        print(' ')
+        for x in donors:
+            print(x)
+        print(' ')
+        f_name = input('Please type the Full Name or list> ')
+    (yes_exist, s) = check_name_exist(f_name, donors)
+    myexit = 0
+    while not myexit == 1:
+        amount = input(s.format(f_name))
+        myexit = check_error(amount)
+    append_donation(f_name, amount, yes_exist, donors)
```

Comment:
it is better to catch your ValueError here and get rid of the `my_exit` var and exit when you catch the exception.